### PR TITLE
Link menu item update

### PR DESCRIPTION
**Pull request by Xinchro**
# Description

Updates the Link Menu styling.
- Adds a bold option to the text of the menu items
- Adds an optional icon to the left side of the menu list

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [ ] I have provided an adequate amount of test coverage
- [x] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have added the documentation
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/docs/welcome/bottom-links-section.component.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { LgSeparatorModule } from '../../separator';
     <lg-link-menu>
       <a *ngFor="let link of links" [href]="link.href" target="_blank">
         <lg-link-menu-item [internal]="false">
-          <lg-link-menu-item-heading>{{ link.text }}</lg-link-menu-item-heading>
+          <lg-link-menu-item-text class="title">{{ link.text }}</lg-link-menu-item-text>
```
[the remark this answers is not in the document]
Comment (Xinchro) [reply projects/canopy/src/lib/docs/welcome/bottom-links-section.component.ts:22]: yep, it is - thanks

**Comment by elenagarrone** [comment]
@Xinchro this seems slightly incorrect:

Screenshot 2023-04-27 at 14 18 06

The input is part of the `text` component 🤔  - I would add it after the description of the `LgLinkMenuItemComponent`. The heading should only be `Inputs`.

File: projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('should contain the default class', () => {
+      expect(fixture.nativeElement.getAttribute('class')).toContain(
+        'lg-link-menu-item-text',
+      );
+    });
```
Comment (elenagarrone) [inline projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.spec.ts:33]: We should add a test for the `isBold` input too

File: projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LgLinkMenuItemTextComponent {
+  @HostBinding('class.lg-link-menu-item-text') class = true;
+  @HostBinding('class.lg-link-menu-item-text--bold') get getIsBold() {
+    return this.isBold;
+  }
+  @Input() isBold = false;
+
+  constructor() {}
```
Comment (elenagarrone) [inline projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.ts:23]: You can remove this constructor

File: projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.spec.ts (hunk 2 above)
Comment (Xinchro) [reply projects/canopy/src/lib/link-menu/link-menu-item-text/link-menu-item-text.component.spec.ts:33]: added 2 tests for this 👍

**Review by elenagarrone — APPROVED** [review]
Nice work 👏

**Review by elenagarrone — APPROVED** [review]

**Review by owensgit — APPROVED** [review]

**Pull request merged**